dotnet/src/VectorDataIntegrationTests/CosmosNoSqlIntegrationTests/Support/CosmosNoSqlTestStore.cs

```diff
@@ -29,16 +27,19 @@ public CosmosClient Client
     public Database Database
         => this._database ?? throw new InvalidOperationException("Call InitializeAsync() first");
 
-    public override VectorStore DefaultVectorStore
-        => this._defaultVectorStore ?? throw new InvalidOperationException("Call InitializeAsync() first");
-
     public CosmosNoSqlVectorStore GetVectorStore(CosmosNoSqlVectorStoreOptions options)
         => new(this.Database, options);
 
     private CosmosNoSqlTestStore()
     {
     }
 
+    protected override void Dispose(bool disposing)
+    {
+        this._client?.Dispose();
```

Comment:
Since the client is instantiated in StartAsync, we should probably dispose it in StopAsync rather than in the Dispose of this class.

Reply:
TBH I did that just to not get another compiler warning (not disposing disposable fields)

Reply:
OK. I think before we just weren't disposing at all, so that warning is good - but it should be OK to just override StopAsync and dispose there, instead of in Dispose? Or does it complain about that too?